05-lists-and-dictionaries/poetry.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2016 Ami . All rights reserved
+
+from random import choice
+import re
+
+class SpeechPart(object):
+    def __init__(self, name, count, vocabulary):
+        self.name = name
+        self.count = count
+        self.vocabulary = vocabulary
+
+    def random_words(self):
+        return self.select_random_words_from_vocabulary()
+
+    def select_random_words_from_vocabulary(self):
+        words = []
+
+        while True:
+            if len(words) == self.count:
```

Comment:
Potentially, this while might run forever (when?), in a normal case we would want to add jere some condition to escape such edge case.